extensions/wikia/Discussions/api/ThreadCreator.class.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+use Swagger\Client\ApiException;
+use Swagger\Client\Discussion\Api\ThreadsApi;
+use Swagger\Client\Discussion\Models\PostInput;
+use Wikia\DependencyInjection\Injector;
+use Wikia\Service\Swagger\ApiProvider;
+
+class ThreadCreator {
+
+	const SERVICE_NAME = "discussion";
+	const TIME_OUT = 5;
+
+	public function create( int $userId, int $siteId, string $body, $title = null ): bool {
+		$postInput = $this->getPostInput( $siteId, $body, $title );
+		$api = $this->getThreadsApi( $userId );
+
+		try {
+			// Currently only creates thread in "General" category of the Discussion.
+			$api->createThread( $siteId, $siteId,  $postInput );
+			$success = true;
+		} catch ( ApiException  $e ) {
```

Comment:
I've found that stringifying the exception `$e` in this way doesn't always yield the best results, giving a generic message rather than the one sent back by the client.  I'd recommend something like:
```php
$response = $e->getResponseObject();
/** @var \Swagger\Client\Discussion\Models\HalProblem $response */
if ( $response instanceof \Swagger\Client\Discussion\Models\HalProblem ) {
	$detail = $response->getDetail();
} else {
	$detail = $e->getMessage();
}
$this->logError( $siteId, $userId, $body, $detail );
```